tests/integration/test_service_accounts.py

```diff
@@ -0,0 +1,596 @@
+    no longer work.
+    """
+    username, namespace = service_account
+    role_name = f"{username}-role"
+    role_binding_name = f"{username}-role-binding"
+
+    # Delete the service account
+    run_service_account_registry(
+        "delete", "--username", username, "--namespace", namespace, "--backend", backend
+    )
+
+    # Check if service account has been deleted
+    service_account_result = subprocess.run(
+        ["kubectl", "get", "serviceaccount", username, "-n", namespace, "-o", "json"],
+        capture_output=True,
+        text=True,
+    )
+    assert service_account_result.returncode != 0
+
+    # Check if the role corresponding to the service account has also been deleted
+    role_result = subprocess.run(
+        ["kubectl", "get", "role", role_name, "-n", namespace, "-o", "json"],
+        capture_output=True,
+        text=True,
+    )
+    assert role_result.returncode != 0
+
+    # Check if the associated role binding has been deleted as well
+    role_binding_result = subprocess.run(
+        [
+            "kubectl",
+            "get",
+            "rolebinding",
+            role_binding_name,
+            "-n",
+            namespace,
+            "-o",
+            "json",
+        ],
+        capture_output=True,
+        text=True,
+    )
+    assert role_binding_result.returncode != 0
+
+    # Check for RBAC permissions, these should be invalid now
+    sa_identifier = f"system:serviceaccount:{namespace}:{username}"
+    rbac_check = subprocess.run(
+        [
+            "kubectl",
+            "auth",
+            "can-i",
+            action,
+            resource,
+            "--namespace",
+            namespace,
+            "--as",
+            sa_identifier,
+        ],
+        capture_output=True,
+        text=True,
+    )
+    assert rbac_check.returncode != 0
+    assert rbac_check.stdout.strip() == "no"
+
+
+@pytest.mark.parametrize("backend", VALID_BACKENDS)
+def test_delete_service_account_that_does_not_exist(namespace, backend):
+    username = str(uuid.uuid4())
+
+    stdout, stderr, ret_code = run_service_account_registry(
+        "delete", "--username", username, "--namespace", namespace, "--backend", backend
+    )
+    assert ret_code != 0
+    assert stdout.strip() == f"Account {username} could not be found."
+
+
+@pytest.mark.parametrize("backend", VALID_BACKENDS)
+def test_service_account_get_primary(namespace, backend):
+    """Test retrieval of primary service account using the CLI.
+
+    Creates an service account with --primary option provided,
+    and then checks whether the same service account is returned
+    upon calling `get-primary` sub-command.
+    """
+    username = str(uuid.uuid4())
+
+    # Create a new service account with --primary option provided
+    run_service_account_registry(
+        "create",
+        "--username",
+        username,
+        "--namespace",
+        namespace,
+        "--backend",
+        backend,
+        "--primary",
+    )
+
```

Comment:
**todo** I've seen that above you were checking for labels. Also here we could check that the service account has the right label, it is a bit more of a low level assertion (white box), but since we have done that above, I would do this here as well

Reply:
The reason that is excluded here is because the `get-primary` command returns with `namespace:username` result only. For checking the labels, the account resource information needs to be fetched from `kubectl get` again, but that test is already done before (i.e. check if created account has appropriate labels).

Reply:
I have seen the check on the MANAGED_BY labels, but not the primary account label. Did I miss that?

Reply:
Ah, you mean the primary account label.. Sorry I misinterpreted that. Yes, that could be checked in addition.. However, `get-primary` would not return the result if that label is not set (because I guess internally that's what `get-primary` uses). Do you think it's still important / relevant to test it explicitly?

Reply:
Yes, but right above you create the account with the primary option. I would expect (hence the assertion) that the account will carry the primary label

It should be one assertion and I believe it is a relevant one (also consistent with the fact that we do check for other labels above)

Reply:
I agree. I've just added an assertion in the `test_create_service_account` method, because that's where I think the best place to check for it would be. As other labels are asserted, the primary account label will be asserted as well (conditionally, which is parameterized so that both primary account and normal account creation are tested).